en/docs/apis/advanced-topics/connect-to-an-external-identity-provider.md

```diff
@@ -0,0 +1,146 @@
+
+![Token exchange flow](../../assets/img/apis/advanced-topics/external-identity-providers/token-exchange-flow.png){.cInlineImage-full}
+
+When you provide a JWT access token (referred to as the `subject_token` in the preceding diagram), a request is sent to the Choreo token endpoint to exchange the JWT with a Choreo JWT. On validation of the request sent, the corresponding IdP configuration is retrieved based on the issuer. Then the `subject_token` validation (signature and expiry time validation) takes place. Successful validation generates a Choreo access token and returns it to you.
+
+Now that you understand the token exchange flow, you can go ahead and add an external identity provider depending on your requirement. 
+
+
+## Add an external identity provider
+
+Follow this procedure to add an external IdP to your Choreo organization:
+
+!!! note
+    You must have organization administrator privileges to add an external IdP.
+
+1. Sign in to the Choreo Console at [https://console.choreo.dev/](https://console.choreo.dev/).
+
+2. In the left navigation menu, click **Settings**.    
+
+3. Go to the **API Management** tab and click **Identity Providers**. This displays the existing identity providers in your organization if you have already added any.
+
+    ![Identity provider list](../../assets/img/apis/advanced-topics/external-identity-providers/identity-provider-list.png){.cInlineImage-full}
+
+4. Click **+ Add Provider**. This displays the identity providers supported by Choreo.
+
+    ![Identity providers](../../assets/img/apis/advanced-topics/external-identity-providers/identity-providers.png){.cInlineImage-full}
+
+5. Click on the IdP you want to add. This displays a form where you must enter details to set up the IdP. <br/> This example walks you through the steps to add **Okta** as the IdP. The details you need to fill in are similar for other IdPs as well.
+
+6. Enter appropriate values for each of the fields and click Next. The following table describes each field in detail: 
+
+    <table>
+    <tr class="header">
+    <th><b>Field</b></th>
+    <th><b>Description</b></th>
+    </tr>
+    <tr class="odd">
+    <td>**Name**</td>
+    <td>The name of the IdP. You cannot modify the value after you add the IdP.</td>
+    </tr>
+    <tr class="even">
+    <td>**Description**</td>
+    <td>A brief description of the IdP.</td>
+    </tr>
+    <tr class="odd">
+    <td>**Allowed Token Audience**</td>
+    <td><b>Okta:</b><br>The audience of the authorization server for which the access token is necessary.<br>
```

Comment:
Okta, Microsoft, Auth0 looks like values at a glance. Do you think we need to specifically say " in the case of Okta:" etc?

Reply:
+1
Added it to go as "if the IdP is Okta:"